main/lsp/requests/hover.cc

```diff
@@ -84,19 +84,13 @@ unique_ptr<ResponseMessage> LSPLoop::handleTextDocumentHover(LSPTypechecker &typ
             string typeString = prettyTypeForMethod(gs, defResp->symbol, nullptr, defResp->retType.type, nullptr);
             response->result = make_unique<Hover>(formatRubyMarkup(clientHoverMarkupKind, typeString, documentation));
         } else if (auto constResp = resp->isConstant()) {
-            const auto &data = constResp->symbol.data(gs);
-            auto type = constResp->retType.type;
-            if (data->isClassOrModule()) {
-                auto singletonClass = data->lookupSingletonClass(gs);
```

Comment:
This `lookupSingletonClass` on the `ConstantResponse` symbol is the same code that `DefLocSaver` does when it creates `ConstantResponse`s, so this whole branch is useless.

Reply:
https://github.com/sorbet/sorbet/blob/master/main/lsp/DefLocSaver.cc#L84-L89